rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -590,63 +515,153 @@ def _compute_default_label_features(
-        else compute a one hot encoding for the label as the feature vector.
+        Moreover, `FeatureArrays` containing the corresponding label ids will be added
+        with subkey `ID`.
+
+        Args:
+            training_data: a training data object holding messages
+            label_attribute: the message attribute which corresponds to the label
+            label_id_dict: a dictionary mapping all relevant labels to some index
+
+        Returns:
+            a RasaDataModel that contains, for each kind of feature (see above),
+            list with one `FeatureArray` per label
+
+        Raises:
+            a `ValueError` if some label is not supported by any training example
         """
-        # Collect one example for each label
-        labels_idx_examples = []
-        for label_name, idx in label_id_dict.items():
-            label_example = self._find_example_for_label(
-                label_name, training_data.intent_examples, attribute
-            )
-            labels_idx_examples.append((idx, label_example))
+        label_indices, label_examples = self._collect_one_example_per_label(
+            messages=training_data.intent_examples, label_id_dict=label_id_dict
+        )
+
+        sequence_features, sentence_features = None, None
 
-        # Sort the list of tuples based on label_idx
-        labels_idx_examples = sorted(labels_idx_examples, key=lambda x: x[0])
-        labels_example = [example for (_, example) in labels_idx_examples]
-        # Collect features, precomputed if they exist, else compute on the fly
-        if self._check_labels_features_exist(labels_example, attribute):
+        # Collect the features from the collected examples, if there are any features
+        all_label_examples_contain_some_features = all(
+            example.features_present(
+                attribute=self.label_attribute,
+                featurizers=self.component_config[FEATURIZERS],
+            )
+            for example in label_examples
+        )
+        if all_label_examples_contain_some_features:
             (
                 sequence_features,
                 sentence_features,
-            ) = self._extract_labels_precomputed_features(labels_example, attribute)
-        else:
+            ) = model_data_utils.combine_attribute_features_from_all_messages(
+                messages=label_examples,
+                attribute=self.label_attribute,
+                featurizers=self.component_config[FEATURIZERS],
+            )
+
+        # If there are no features or if the sentence features are missing but
+        # required, then load the default label (sentence) features.
+        needs_sentence_features = self._needs_sentence_features_for_labels()
+        if not sentence_features and (not sequence_features or needs_sentence_features):
+
+            if sequence_features and not sentence_features and needs_sentence_features:
```

Comment:
You can simplify this second `if` too with:

```
if sequence_features and needs_sentence_features:
```

Reply:
That was supposed to be more readable but apparently it was just adding confusion ^^. Will shorten that then